loopy/__init__.py

```diff
@@ -487,6 +487,48 @@ def make_copy_kernel(new_dim_tags, old_dim_tags=None):
 # }}}
 
 
+# {{{ einsum
+
+def make_einsum(spec, arg_names, knl_creation_kwargs=None):
+    arg_spec, out_spec = spec.split("->")
```

Comment:
sanity checks on out_spec: All out indices must be unique

Reply:
@thomasgibson Could you take care of this, too?

Reply:
Done! https://github.com/inducer/loopy/pull/328/commits/b1bb749d80b6c5d06927de9f17b5af421d5a3a89